realm/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -493,8 +493,59 @@ public TableView findAll() {
         }
     }
 
-    protected native long nativeFindAll(long nativeQueryPtr, long start, long end, long limit);
+    // Suppose to be called from the a background SharedGroup thread
+    public long findAllWithHandover(long bgSharedGroupPtr) {
+        validateQuery();
+
+        // Execute the disposal of abandoned realm objects each time a new realm object is created
+        context.executeDelayedDisposal();
+        return nativeFindAllWithHandover(bgSharedGroupPtr, nativePtr, 0, Table.INFINITE, Table.INFINITE);
+        // handover nativeViewPtr to UI Thread
+
+    }
+
+    /**
+     * This will use the background shared group to import the query pointer from the handover one
```

Comment:
This could use a rewrite to be more clear, perhaps

"This will use the background SharedGroup to import the query (using the handover object), Run the query, and return the table view to the caller SharedGroup using the handover object."